### ISSUE TITLE
How should i understand  useing  self conditioning in DiffusionPriorNetwork which is Hinton's group's new ddpm technique.

### ISSUE THREAD

**Issue by SELECT-FROM** [lucidrains/DALLE2-pytorch#273]
If self.self_cond is true , how should I understand learned_queries .

        if self.self_cond:
            learned_queries = torch.cat((image_embed, self_cond), dim=-2)

**Comment by lucidrains** [comment]
@SELECT-FROM that doesn't look right :disappointed: i believe it should be correct now

the logic should have been to include the self conditioning token just before the learned queries

**Comment by SELECT-FROM** [comment]
> @SELECT-FROM that doesn't look right 😞 i believe it should be correct now
> 
> the logic should have been to include the self conditioning token just before the learned queries
Thank you for your reply. I have another question is in what scenario or requirement should self conditioning be set to true? The default value false is always used in the init method of DiffusionPriorNetwork

**Comment by SELECT-FROM** [comment]
@lucidrains Thank you for your reply. 
I have another question is in what scenario or requirement should self conditioning be set to true? The default value false is always used in the init method of DiffusionPriorNetwork

**Comment by lucidrains** [comment]
@SELECT-FROM the whole field is very empirical

i would recommend just running a comparison test with that one flag `self_cond` toggled
